code/datums/elements/decals/_decal.dm

```diff
@@ -3,14 +3,60 @@
 	id_arg_index = 2
 	var/cleanable
 	var/description
+	/// If true this was initialized with no set direction - will follow the parent dir.
+	var/directional
 	var/mutable_appearance/pic
 
-/datum/element/decal/Attach(atom/target, _icon, _icon_state, _dir, _cleanable=FALSE, _color, _layer=TURF_LAYER, _plane=FLOOR_PLANE, _description, _alpha=255)
+/// Remove old decals and apply new decals after rotation as necessary
+/datum/controller/subsystem/processing/dcs/proc/rotate_decals(datum/source, old_dir, new_dir)
+	SIGNAL_HANDLER
+
+	if(old_dir == new_dir)
+		return
+	var/list/resulting_decals_params = list() // param lists
+	var/list/old_decals = list() //instances
+
+	if(!source.comp_lookup || !source.comp_lookup[COMSIG_ATOM_UPDATE_OVERLAYS])
+		//should probably also unregister itself
+		return
+
+	switch(length(source.comp_lookup[COMSIG_ATOM_UPDATE_OVERLAYS]))
```

Comment:
What is this for? Are you sure this is sane? It's scary to me to be using internal stuff like this.